Use AndroidX, Gradle 8.2, AGP 8.2.2, protobuf 3.25.2 and several nitpicky things

weebl2000 opened the pull request:
No clue if this is wanted, but my Android Studio was warning me a lot and this seems to work.

roussosalex commented:
Thanks for contributing (again)! Could you please rebase this PR on the updated `v2` branch?

Commit:
Gradle 8.2 and AGP 8.2.2

Commit:
Gradle wrapper 8.2

Updated wrapper with the wrapper plugin:

./gradlew wrapper --gradle-version=8.2 --distribution-type=bin

Commit:
Upgrade protobuf library to latest release

Commit:
Use androidX libraries

Commit:
Migrate buildconfig

Commit:
Fix missing dependencies

Commit:
nitpick " -> '

For non-variable strings

weebl2000 commented:
> Thanks for contributing (again)! Could you please rebase this PR on the updated `v2` branch?

Sure, I've reapplied the suggestions on the latest v2 branch.

And nitpicked a little bit on the build files to make them a little more consistent.

Did not test all functionality extensively but it seems to build and works on my Pixel 6 running QPR 14 / 3.1(AP11.231215.009)

Using AndroidX packages removes the need for the lint-baseline.xml - so I've gone ahead and removed that.

Commit:
Remove lint-baseline.xml since we use AndroidX now

roussosalex commented on app/src/main/assets/html/desfire-info.en.html:
```diff
@@ -31,4 +31,4 @@ <h5>Technical details</h5>
 </body>
-</html>
\ No newline at end of file
+</html>
```
Fixing newlines and quotes should be in a separate PR

weebl2000 replied:
Done, removed any files that only have newlines added and reverted all lines that only had quotation marks changed.

roussosalex reviewed (changes requested):
Much needed update to AndroidX. Thanks!

weebl2000 committed:
Revert unchanged lines other than quotation marks
